compiler/testData/codegen/box/extensionFunctions/contextReceivers/fromKEEP/functionalType.kt

```diff
@@ -0,0 +1,18 @@
+// !LANGUAGE: +ContextReceivers
+// TARGET_BACKEND: JVM_IR
+// IGNORE_BACKEND_FIR: JVM_IR
+
+class Param
+class O {
+    val o = "o"
+}
+class K {
+    val k = "k"
+}
+
+context(O)
+fun <T> K.f(g: context(O) K.(Param) -> T) = g(this@O, this@K, Param())
```

Comment:
Please, add a test (diagnostic or box) that would involve context receivers in different order as discussed privately (`context(A, B) <...>` used where `context(B, A) <...>` expected).

If we actually allow it, then someone should determine a proper context receiver transposition (I'd assume FE should do this anyway when type checking such code, and it should be a single point of truth for everybody else), and PSI2IR should generate a wrapper function performing such transposition. Should also think about equality for resulting objects.

In other words, I wouldn't be surprised if we prohibit such code.

Reply:
For now, let's prohibit such code. Added a test in [the commit](https://github.com/JetBrains/kotlin/pull/4488/commits/8970a97b456b1e177058abb38fbe990ab8613d3d).
It looks like it will be much cheaper just to add it as a limitation in KEEP.